modular_skyrat/modules/powerarmor/code/powerarmor.dm

```diff
@@ -0,0 +1,839 @@
+		upgradelimit -= upgrade_item.upgrade_cost
+		update_upgrades()
+		return
+	if(istype(W, /obj/item/assembly/signaler/anomaly))
+		if(upgradeboosted)
+			return
+		upgradeboosted = TRUE
+		to_chat(user, span_notice("You use [W] to boost [src] upgrade capacity!"))
+		upgradelimit += 10
+		qdel(W)
+		return
+	if(W.tool_behaviour == TOOL_SCREWDRIVER)
+		W.play_tool_sound(src, 50)
+		if(!do_after(user, 10 SECONDS, target = wearer))
+			return
+		upgradelimit = 20
+		if(upgradeboosted)
+			upgradelimit = 30
+		for(var/obj/check_contents in contents)
+			if(istype(check_contents, /obj/item/powerarmor_upgrade))
+				check_contents.forceMove(get_turf(src))
+		armor_upgraded = list(0, 0, 0, 0, 0, 0, 0)
+		healing_upgraded = list(FALSE, FALSE, FALSE, FALSE, FALSE)
+		speed_upgraded = 0
+		misc_upgraded = list(FALSE, FALSE, FALSE, FALSE, FALSE)
+		update_upgrades()
+		W.play_tool_sound(src, 50)
+	return ..()
+
+/obj/item/clothing/head/hooded/powerarmor
+	name = "power armor helmet"
+	desc = "A helment for the power armor that is capable of being modified to give the user even greater potential."
+	icon = 'modular_skyrat/modules/powerarmor/icons/hats.dmi'
+	worn_icon = 'modular_skyrat/modules/powerarmor/icons/head.dmi'
+	icon_state = "helmet0"
+	mutant_variants = NONE
+
+	flags_inv = HIDEMASK|HIDEEARS|HIDEEYES|HIDEFACE|HIDEHAIR|HIDEFACIALHAIR|HIDESNOUT
+	flags_cover = HEADCOVERSEYES | HEADCOVERSMOUTH | PEPPERPROOF
+	flash_protect = FLASH_PROTECTION_NONE
+
+/obj/item/clothing/head/hooded/powerarmor/Initialize()
+	. = ..()
+	if(suit && istype(suit, /obj/item/clothing/suit/hooded/powerarmor))
+		update_upgrades()
+
+/obj/item/clothing/head/hooded/powerarmor/proc/update_upgrades()
+	if(!suit)
+		return
+	if(!istype(suit, /obj/item/clothing/suit/hooded/powerarmor))
+		return
+	var/obj/item/clothing/suit/hooded/powerarmor/power_suit = suit
+	armor.melee = power_suit.armor_upgraded[1] * 20
+	armor.bullet = power_suit.armor_upgraded[2] * 20
+	armor.laser = power_suit.armor_upgraded[3] * 20
+	armor.energy = power_suit.armor_upgraded[4] * 20
+	armor.bio = power_suit.armor_upgraded[5] * 20
+	armor.rad = power_suit.armor_upgraded[6] * 20
+	armor.fire = power_suit.armor_upgraded[7] * 20
+	clothing_flags = initial(clothing_flags)
+	min_cold_protection_temperature = initial(min_cold_protection_temperature)
+	max_heat_protection_temperature = initial(max_heat_protection_temperature)
+	heat_protection = initial(heat_protection)
+	cold_protection = initial(cold_protection)
+	if(power_suit.misc_upgraded[1])
+		heat_protection = HEAD
+		cold_protection = HEAD
+		clothing_flags = STOPSPRESSUREDAMAGE | THICKMATERIAL
+		max_heat_protection_temperature = SPACE_HELM_MAX_TEMP_PROTECT
+		min_cold_protection_temperature = SPACE_HELM_MIN_TEMP_PROTECT
+	flash_protect = FLASH_PROTECTION_NONE
+	if(power_suit.misc_upgraded[3])
+		flash_protect = FLASH_PROTECTION_WELDER
+
+/obj/item/powerarmor_upgrade
+	name = "power armor upgrade"
+	desc = "A small item that will upgrade the power suit"
```

Comment:
grammar

Reply:
whats the issue?

Reply:
it kinda sounds weird in general but at least there should be a period at the end

Reply:
ah